### PR TITLE
pr: metric.py

### PR DESCRIPTION
**Pull request by zhangjiekui**
(1) unified returned value to 'None' if deque is empty
(2) by float(torch.tensor), fixed bug: for np.array(list(self._deque)) in self.median, self.avg, np.array(list(torch.Tensor)) will raise exception
(3) added comments accordingly

### CODE REVIEW

**Comment by FateScript** [comment]
First please lint your code. 
When will you use empty queue? It should not happen accroding to our design.
